aws/external/provider.go

```diff
@@ -343,3 +343,13 @@ func NewWithEC2MetadataRegion(ctx context.Context, client *ec2metadata.Client) W
 func (p WithEC2MetadataRegion) GetRegion() (string, error) {
 	return p.client.Region(p.ctx)
 }
+
+// WithS3UseARNRegion is used to specify whether to enable usage of the region specified
+// in an ARN passed as an S3 bucket value.
+type WithS3UseARNRegion bool
+
+// GetS3UseARNRegion returns whether to enable usage of the region specified
+// in an ARN passed as an S3 bucket value.
+func (w WithS3UseARNRegion) GetS3UseARNRegion() (bool, error) {
+	return bool(w), nil
+}
```

Comment:
Does this need to be exported in the `external` package or unexported within the s3 client package? This provider probably can be generated.